### ISSUE TITLE
GitHub actions workflows that use master branch fail when using npm version 7.6.0

### ISSUE THREAD

**Issue by fluiddot** [WordPress/gutenberg#29528]
## Description

When using npm version `7.6.0` in the workflows `Compressed Size / Check (pull_request)` and `Performances Tests / Run performance tests (pull_request)`, we checkout `master` but due to the [recent changes limiting the version of npm to v6](https://github.com/WordPress/gutenberg/pull/29204), the processes fails.

## Step-by-step reproduction instructions

1. Create a test branch
2. Install npm version 7.6.0
3. Remove `<7` condition from the npm engine section of [`package.json`](https://github.com/WordPress/gutenberg/blob/581121cb2d3a40fb34073f83219033d92e2d02f0/package.json#L19) file ([example](https://github.com/WordPress/gutenberg/pull/28886/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L19))
3. Run `npm install`
4. Add the following code to a GitHub actions workflow that uses `master` branch (like [`Compressed Size`](https://github.com/WordPress/gutenberg/blob/trunk/.github/workflows/bundle-size.yml)):
```
    - name: Use Node.js 14.x
      uses: actions/setup-node@v1
      with:
        node-version: 14.x

    - name: Install latest NPM version
      run: npm install -g npm@7.6.0
```
_The code has to be placed after the checkout and npm cache restore, here is an [example](https://github.com/WordPress/gutenberg/pull/28886/files#diff-749feccfd42652f4a7571f0103a5b5b516b2b6d40f64f64200ac4c64870342aaR15-R22)._
5. Push the changes
6. Create a test PR
7. Observe that the modified workflow is failing

## Expected behaviour

GitHub actions workflows that use `master` branch should succeed.

## Actual behaviour

GitHub actions workflows that use `master` branch fail.

## Screenshots or screen recording (optional)

N/A

## WordPress information
- WordPress version: N/A
- Gutenberg version: 10.0.1
- Are all plugins except Gutenberg deactivated? N/A
- Are you using a default theme (e.g. Twenty Twenty-One)? N/A

## Device information
- Device: N/A
- Operating system: N/A
- Browser: N/A

**Comment by ockham** [comment]
This one can be simply fixed by 

```diff
diff --git a/package.json b/package.json
index 3c75522414..f3c07a917c 100644
--- a/package.json
+++ b/package.json
@@ -16,7 +16,7 @@
        },
        "engines": {
                "node": ">=10.0.0",
-               "npm": ">=6.9.0 <7"
+               "npm": ">=6.9.0"
        },
        "config": {
                "GUTENBERG_PHASE": 2,
```

or, if we prefer,

```diff
diff --git a/package.json b/package.json
index 3c75522414..dcbe81d1c2 100644
--- a/package.json
+++ b/package.json
@@ -16,7 +16,7 @@
        },
        "engines": {
                "node": ">=10.0.0",
-               "npm": ">=6.9.0 <7"
+               "npm": ">=7.6.0"
        },
        "config": {
                "GUTENBERG_PHASE": 2,
```

, right?

It's just that we can't do that yet because of #29524 and 29527, right?

**Comment by fluiddot** [comment]
Yeah, this could be easily fixed just by changing the npm engine value as you commented.
 
The only thing to keep in mind is that we would need to merge this change into `trunk` branch first so the PR checks of  ["Update lock file format version to version 2 (NPM 7)" PR](https://github.com/WordPress/gutenberg/pull/28886) pass.

**Comment by Mamaduka** [comment]
Engines update would be nice.

Currently can't install dependencies on Apple Silicon machines because of this restriction.

**Comment by youknowriad** [comment]
We've updated npm a long time ago, this is not valid anymore.
